rfcs/20251105-rms-norm-graph-op/README.md

```diff
@@ -0,0 +1,321 @@
+  the back. For example, `axis=-1` normalizes over the last dimension only.
+
+- **epsilon** (optional, default=1e-5): Small constant added to denominator for
+  numerical stability.
+
+- **dst** (required): Output tensor with the same shape as src.
+
+- **rrms** (optional): Reciprocal of RMS (`1/RMS(x)`). Can be cached for reuse
+  in backward pass to avoid redundant computation. Shape is the same as src with
+  normalized dimensions reduced to size 1.
+
+**Semantic Notes:**
+
+1. Gamma parameter is optional. If not provided, only normalization is performed
+   without scaling (equivalent to gamma=1).
+2. No beta/bias parameter (distinguishes from LayerNorm).
+3. The `rrms` output is optional and typically used when backward pass is needed.
+   It stores the reciprocal of RMS for each sample, enabling efficient gradient
+   computation without recomputing RMS.
+
+**Example Usage:**
+
+```cpp
+// Create RMSNorm op for transformer layer
+// Input: [batch=32, seq_len=128, hidden_dim=768]
+// Normalize over last dimension
+auto rms_norm_op = graph::op_t(graph::op::kind::RMSNorm);
+rms_norm_op.set_attr<int64_t>(graph::op_attr::axis, -1);
+rms_norm_op.set_attr<float>(graph::op_attr::epsilon, 1e-5f);
+```
+
+**Pros:**
+
+1. Clear semantics with one-to-one mapping to framework operations.
+2. Simpler parameter set than LayerNorm (no mean, no beta).
+3. Distinguishes RMSNorm from LayerNorm at the API level, making fusion
+   patterns explicit.
+4. Easier for frameworks to integrate.
+
+**Cons:**
+
+1. Adds another normalization op to maintain.
+2. Requires separate fusion patterns from LayerNorm.
+
+### Option 2: Extend LayerNorm with a flag to indicate RMSNorm behavior
+
+| LayerNorm          | oneDNN Graph API   |
+| ------------------ | ------------------ |
+| input              | src                |
+| input              | gamma (optional)   |
+| input              | beta (optional)    |
+| attribute          | keep_stats         |
+| attribute          | begin_norm_axis    |
+| attribute          | use_affine         |
+| attribute          | epsilon            |
+| attribute          | rms_norm           |
+| output             | dst                |
+| output             | mean (optional)    |
+| output             | variance (optional)|
+
+Add a new attribute `rms_norm` (default=false) to LayerNorm. When `rms_norm=true`,
+the operation performs RMS normalization (no mean subtraction, no variance computation).
+The `beta` parameter is ignored and `mean`/`variance` outputs are not generated when
+`rms_norm=true`.
+
+**Pros:**
+
+1. Reuses existing LayerNorm infrastructure.
+2. Fewer operations to maintain.
+3. Single operation covers both LayerNorm and RMSNorm use cases.
+
+**Cons:**
+
+1. Overloaded semantics - LayerNorm becomes ambiguous.
+2. Framework integration is less intuitive (need to set flags).
+
+## Decision
+
+**Choose Option 1** - Add RMSNorm as a dedicated operation.
```

Comment:
@TaoLv, could you, please, tag engineers responsible for PyTorch integration to verify their preferences?